src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,158 @@
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useState, useCallback } from 'react';
+import { getTodos, getFilteredTodos } from '../../api';
+import { Todo } from '../../types/Todo';
+import { Selected } from '../../types/Enum';
+
+type Props = {
+  setTodos: (value: Todo[]) => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  setTodos,
+}) => {
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [value, setValue] = useState('');
+
+  const getIncludesTodos = useCallback(
+    (val: string, currTodos: Todo[]) => {
+      return currTodos.filter((todo) => todo.title.includes(val));
+    },
+    [],
+  );
+
+  // const handleFilterChange = async (
+  //   selectedValue: string,
+  //   inputValue: string,
+  // ) => {
+  //   setSelectedOption(selectedValue);
+
+  //   let filteredTodos;
+
+  //   switch (selectedValue) {
+  //     case Selected.All:
+  //       filteredTodos = await getTodos();
+  //       break;
+  //     case Selected.Active:
+  //       filteredTodos = await getFilteredTodos(false);
+  //       break;
+  //     case Selected.Completed:
+  //       filteredTodos = await getFilteredTodos(true);
+  //       break;
+  //     default:
+  //       filteredTodos = await getTodos();
+  //   }
+
+  //   if (inputValue) {
+  //     filteredTodos = getIncludesTodos(inputValue, filteredTodos);
+  //   }
+
+  //   setTodos(filteredTodos);
+  // };
+
+  // const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+  //   const searchValue = e.target.value;
+
+  //   setValue(searchValue);
+  //   handleFilterChange(selectedOption, searchValue);
+  // };
+
+  // const handleSelectChange = async (
+  //   event: React.ChangeEvent<HTMLSelectElement>,
+  // ) => {
+  //   const selectedValue = event.target.value;
+
+  //   handleFilterChange(selectedValue, value);
+  // };
+
+  const handleFilterChange = useCallback(
+    async (selectedValue: string, inputValue: string) => {
+      setSelectedOption(selectedValue);
+
+      let filteredTodos;
+
+      switch (selectedValue) {
+        case Selected.All:
+          filteredTodos = await getTodos();
+          break;
+        case Selected.Active:
+          filteredTodos = await getFilteredTodos(false);
+          break;
+        case Selected.Completed:
+          filteredTodos = await getFilteredTodos(true);
+          break;
+        default:
+          filteredTodos = await getTodos();
+      }
+
+      if (inputValue) {
+        filteredTodos = getIncludesTodos(inputValue, filteredTodos);
+      }
+
+      setTodos(filteredTodos);
+    },
+    [getIncludesTodos, setTodos],
+  );
+
+  const handleInputChange = useCallback((e: React
+    .ChangeEvent<HTMLInputElement>) => {
+    const searchValue = e.target.value;
```

Comment:
I think we all know what is `e` means, but try to use more meaningful names like `event`